### PR TITLE
Made patches limb based

### PR DESCRIPTION
**Pull request by Razharas**
Added proc to humans to check cover of anything, can be useful
Added new proc to reagents that is triggered by patches on organs
exclusively
Added var to all limbs to define maximum amount of patches, currently 3
Thats change of mechanic, so balance while playing a role wasnt the
goal, im open to suggestions, tweaks and changes(in sane boundaries)
Theres also half a year old commit sneaked into this pr but done nothing, im not sure why, git is magic

### CODE REVIEW

**Comment by pudl** [comment]
my suggestion is you close this pr

**Comment by WJohn** [comment]
I don't really understand the use of patches then... They're like 1-time-use time consuming non stackable bruise/ointment packs?

**Comment by Razharas** [comment]
Yes

**Comment by WJohn** [comment]
So... they're literally just worse? Why even have these? Why such a gigantic nerf from what was had before? Simply making it take time to apply it to a limb is enough of a nerf as-is when that was done on NT.

**Comment by Razharas** [comment]
You mean worse than bruisepacks? I thought they are removed

**Comment by WJohn** [comment]
No those still exist in the code and you can find an ointment and bruise pack in the dorm room. But seriously whether those things exist anymore or not doesn't matter, it's way WAY worse than what we had in the past.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents.dm [hunk 1]
```diff
@@ -62,6 +62,10 @@ datum/reagent/proc/reaction_mob(var/mob/M, var/method=TOUCH, var/volume) //By de
 					M.reagents.add_reagent(self.id,self.volume/2)
 	return 1
 
+datum/reagent/proc/on_touch_apply(var/obj/item/organ/limb/L)
+	var/mob/living/carbon/human/H = L.owner
+	holder.remove_reagent(src.id, metabolization_rate * H.metabolism_efficiency)
```
Comment (Iamgoofball) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents.dm]: Hold up here.

patches are supposed to transfer the reagents into the body.

**Comment by Iamgoofball** [comment]
I'm not too certain I agree with some of these nerfs. I'll think it over.

**Comment by TZK13** [comment]
I don't see why removing bruise packs and ointments from medical sources was even necessary, they certainly weren't connected to chemistry. If the intention with the new chemistry was to make medical treatment a more involved process that in this case it achieved the opposite effect as currently an MD doesn't need to know which of the patient's limbs are injured, they just have to know what damage type it is and use the corresponding patch. This PR brings patches more inline with how bruise pack and ointments work but that means that they're essentially the worst aspect of pills, bruise packs, and ointments combined as WJohn said.

**Comment by Deantwo** [comment]
Making it so that the patches' frist first/instant effect is only appiled to the target limb, and then have the rest of the reagents transfered to the mob would be more correct.l
That way we would have more control over what limb is healed most. But still giving the reagents so some other wounds heal a little too.

But really I thought @Iamgoofball was working on this in https://github.com/tgstation/-tg-station/pull/7382 too.

**Comment by Fox-McCloud** [comment]
Patches heal ~64 on average--20 up front and an average of ~44 over time (provided you're using medkit ones) and you don't have to target limbs with them; that's their advantage (heal over time+non-targeted); applying these nerfs makes them strictly worse in just about every way.

Also, as goofball said, patches apply touch and transfer to the mob; this is 100% intentional and removing it pretty much makes patches 100% worthless.

**Comment by TZK13** [comment]
Not having to target limbs with them makes being an MD even more mind numbingly simplistic though which is why bruise packs and ointments should be returned as those actually require more effort to utilize.

**Comment by pudl** [comment]
goonchem making simple healing mindless compliments the fact that goonmed has different surgery, cryo, and sleepers.
i tried to port goonmed and people got upset about it, so now we're left with a bastardized chem that should be fucking reverted already. don't bog down goonchem until it's on star trek chem's pathetic level, that's an insult to an originally fine system.

**Comment by Deantwo** [comment]
Not sure how clear my last comment was. Let me try again.

I think that only the *TOUCH* effect could be limb based.
But the (remaining) reagents should be tranfered to the mob.
Having the whole patch be limb based would be sad.

Currtently patches are pretty much the same as pills.
And in the end limb based healing only really matters for like 2 or 3 chemicals.

**Comment by paprka** [comment]
goonchem shouldn't need goonmed to stand up on its own, patches fill the niche of applying touch based reactions and nothing more. I see no reason for them to have anything to do with the rest of the medical system.

I will state it again, this time in bold letters:

**Stop shoving goonchem into the rest of the medical system you fucking morons.**

You're making an already bad problem worse. Worry about fixing goonchem to be good first before turning patches into pseudo bruise packs as a replacement. Bruise packs and ointment are better gameplay than patches EVER will be.

**Comment by pudl** [comment]
goonchem was specifically tailored to goonmed, just as star trek chem had been tailored to our med. the biggest reason that goonchem is going to ultimately fail is because you refuse to allow goonchem to integrate with our med and make them work together. pre-goonmed, star trek chem's big dick was all in medical. you only have a problem with it now because of your grudge against goonchem.

**Comment by paprka** [comment]
hey quilty i speak for literally everyone on the server

f u c k o f f
u
c
k
o
f
f

**Comment by pudl** [comment]
way to be a team player

**Comment by paprka** [comment]
Wow it's almost like I planned on fixing that by adding defibs, bleeding, removing sleepers...

**Comment by pudl** [comment]
all three of those things make no improvements whatsoever.
defibs in general are just a "makes life easier" machine when, regardless of if you're like a mile away from medbay, people are usually dragged to medbay anyway for mannitol. players just don't carry around syringes of it, don't ask me why.
bleeding is on goonstation too, in fact bay's bleeding system was originally conceived as a rather huge knock off of goon's (it was identical in every way) but they changed it quite a bit to suit to their server with iv drips, etc.
removing sleepers would have made no change whatsoever. they were never used ever, and now that you're removing them post-goon chem you're going to just upset a lot of people, since they're used incredibly often to avoid the use of goonchem.

**Comment by paprka** [comment]
"since they're used incredibly often to avoid the use of goonchem."

BECAUSE

YOU FAGGOTS

REMOVED

THE

BRUISE PACKS AND OINTMENTS

**Comment by pudl** [comment]
the problem wouldn't be nearly as bad if you would allow styptic patches in medkits and goof's buff would get merged, since that would make patches 100% accurate to goon's, which is basically the same as brute patches and ointment.

**Comment by paprka** [comment]
holy shit patches are such bad gameplay how can you not see this

we aren't goon, go away

**Comment by pudl** [comment]
goon's medbay does not revolve around chemistry, because medbay on goon is a lot more indepth than our medbay.
our health system is literal garbage. our surgery is garbage. everything about mds is garbage. goonchem isn't making it worse, it's just exposing the cracks in a long broken system.

**Comment by dannno** [comment]
>holy shit patches are such bad gameplay how can you not see this

yeah kinda.
it's not exposing cracks, it's making cracks that should have been anticipated and filled via testing before it was merged.

**Comment by paprka** [comment]
^

**Comment by Iamgoofball** [comment]
I'm okay with this as long as patches still transfer reagents, as certain chems are balanced around patches transferring reagents. Until then, don't merge this please.

**Comment by Deantwo** [comment]
Can we move this discussion to the forum too maybe? Post a link here when/if you make a thread.

File: code/modules/reagents/reagent_containers/patch.dm [hunk 2]
```diff
@@ -13,13 +13,42 @@
 	..()
 	icon_state = "bandaid" // thanks inheritance
 
-/obj/item/weapon/reagent_containers/pill/patch/afterattack(obj/target, mob/user , proximity)
-	return // thanks inheritance again
+/obj/item/weapon/reagent_containers/pill/patch/attack(mob/living/carbon/human/M, mob/user)
+	if(istype(M))
+		var/obj/item/organ/limb/affected = canconsume(M, user)
+		if(affected && affected.status == ORGAN_ORGANIC)
+			user.visible_message("<span class='notice'>[user] manages to [apply_method] [src] on [M].</span>", \
```
Comment (Iamgoofball) [inline code/modules/reagents/reagent_containers/patch.dm]: But what if I want to apply a facid patch to someone's robotic arm to melt it?

**Comment by Razharas** [comment]
Currently goof is consulting me on chems, so most likely some/all parts of this besides cover checks are going to change

File: code/modules/reagents/reagent_containers/patch.dm (hunk 2 above)
Comment (Razharas) [inline code/modules/reagents/reagent_containers/patch.dm]: Currently only 3 healing chems have any effect, anything else will do exactly nothing
Yes i know its bad, yes this will most likely be changed

**Comment by Iamgoofball** [comment]
After serious consideration and multiple failed code attempts, the code required to restore the missing features, unbreak the balance of touchchems, and not make patches utterly worthless is so snowflake and stupid we might as well ditch limb patches until dismemberment rolls around.

Plus, this will make the players even more pissed since they see chem as nerfed into the gourd already.

:-1:
